### PR TITLE
Create StagedBitSet

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

**Comment by danking** [comment]
@jigold added

File: src/test/scala/is/hail/asm4s/StagedBitSetSuite.scala [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    val fb = functionBuilder[Boolean]
+    val bs = new StagedBitSet(fb)
+    val x = bs.newBit()
+    val y = bs.newBit()
+    fb.emit(f(x, y))
+    fb.result()()()
+  }
+
+  @Test
+  def testSetOneBit() {
```
Comment (cseed) [inline src/test/scala/is/hail/asm4s/StagedBitSetSuite.scala]: You might consider a test with more than 64-bits.

File: src/main/scala/is/hail/asm4s/StagedBitSet.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import scala.collection.generic.Growable
+
+class StagedBitSet(fb: FunctionBuilder[_]) {
+  private var used = 0
+  private var bits: LocalRef[Long] = null
+  private var count = 0
+
+  def newBit(): SettableBit = {
+    if (used >= 64 || bits == null) {
+      bits = fb.newLocal[Long]("bitset"+count)
```
Comment (cseed) [inline src/main/scala/is/hail/asm4s/StagedBitSet.scala]: You're using a `newLocal` that doesn't exist in this version.  Remove the string here and put it back in the PR that puts the `newLocal` with a base name in?

File: src/main/scala/is/hail/asm4s/StagedBitSet.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+class StagedBitSet(fb: FunctionBuilder[_]) {
+  private var used = 0
+  private var bits: LocalRef[Long] = null
+  private var count = 0
+
+  def newBit(): SettableBit = {
+    if (used >= 64 || bits == null) {
+      bits = fb.newLocal[Long]
+      count += 1
+      fb.emit(bits.store(0L))
```
Comment (cseed) [inline src/main/scala/is/hail/asm4s/StagedBitSet.scala]: I'm going to approve this, but this seems lightly dangerous, although I'm not sure what else to do, except maybe have `newBit` return a pair including an initialization code.  Problem I'm imagining:

```
  emit if
    val mx = sbs.newBit() # first bit in word
    ...
  end if
  val my = sbs.newBit() // second bit in long, uninitialized!
```

**Pull request merged**